components/data-bridge/org.wso2.carbon.databridge.agent/src/main/java/org/wso2/carbon/databridge/agent/endpoint/DataEndpointGroup.java

```diff
@@ -135,9 +135,14 @@ private void tryPut(Event event, long timeoutMS) throws EventQueueFullException
                     this.ringBuffer.publish(sequence);
                     break;
                 } catch (InsufficientCapacityException ex) {
-                    if (stopTime > System.currentTimeMillis()) {
+                    if (stopTime <= System.currentTimeMillis()) {
                         throw new EventQueueFullException("Cannot send events because the event queue is full", ex);
                     }
+                    try {
+                        Thread.sleep(1);
+                    } catch (InterruptedException e) {
+
```

Comment:
@sinthuja @mohanvive 

Won't the thread enter a busy wait without doing any actual work if there is no Thread.sleep() or Thread.yield in this situation? Otherwise, this thread might run on a tight loop and occupy CPU without letting other threads, such as the disruptor consumer threads, use the CPU cores. IMHO, during an InsufficientCapacityException, it would be better if the producer thread can go to sleep and try to give much chance as possible for Consumer threads to run. WDYT?